### ISSUE TITLE
Config schemas are enforced for disabled rules

### ISSUE THREAD

**Issue by not-an-aardvark** [eslint/eslint#7690]
**Tell us about your environment**

* **ESLint Version:** master
* **Node Version:** 7.2.0
* **npm Version:** 3.10.9

**What parser (default, Babel-ESLint, etc.) are you using?**

default

**Please show your full configuration:**

(n/a)

**What did you do? Please include the actual source code causing the issue.**

I created a rule with a required schema, like this:

```js
// "example-rule"
module.exports = {
    meta: {
        schema: {
            // At least one option must be provided
            type: "array",
            minItems: 1
        }
    },
    create(context) {}
};
```

I ran the rule on some code in `foo.js` with three different sets of options:

```bash
eslint foo.js --rule 'example-rule: [error, 4]' # case 1
eslint foo.js --rule 'example-rule: error'      # case 2
eslint foo.js --rule 'example-rule: off'        # case 3
```

**What did you expect to happen?**

In Case 1, I expected ESLint to run to completion without reporting an error, because a valid schema was provided.

In Case 2, I expected ESLint to report a schema error, because an invalid schema was provided (it was missing an option).

In Case 3, I expected ESLint to run to completion without reporting an error. The rule is disabled, so I shouldn't have to provide a schema while disabling it (e.g. `example-rule: [off, 4]`)

**What actually happened? Please include the actual, raw output from ESLint.**

ESLint worked as expected in Case 1 and Case 2, but it reported a schema error in case 3.

```
eslint foo.js --rule 'example-rule: off'
CLI:
	Configuration for rule "example-rule" is invalid:
	Value "" has less items than allowed.

Error: CLI:
	Configuration for rule "example-rule" is invalid:
	Value "" has less items than allowed.

    at Object.validateRuleOptions (/path/to/eslint/lib/config/config-validator.js:115:15)
    at CLIEngine.Object.keys.forEach.name (/path/to/eslint/lib/cli-engine.js:476:19)
    at Array.forEach (native)
    at new CLIEngine (/path/to/eslint/lib/cli-engine.js:475:43)
    at Object.execute (/path/to/eslint/lib/cli.js:169:28)
    at process.stdin.pipe.text (/path/to/eslint/bin/eslint.js:59:32)
    at ConcatStream.<anonymous> (/path/to/eslint/node_modules/concat-stream/index.js:36:43)
    at emitNone (events.js:91:20)
    at ConcatStream.emit (events.js:185:7)
    at finishMaybe (/path/to/eslint/node_modules/concat-stream/node_modules/readable-stream/lib/_stream_writable.js:475:14)

```
